sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -549,15 +549,18 @@ class HDFSFileCatalog(
 
   refresh()
 
+  private def isInvalidFile(f: FileStatus): Boolean =
+    f.getPath.getName.startsWith("_") || f.getLen == 0
```

Comment:
Yeah, its not clear to me that we want to skip 0 sized files.  They really shouldn't be there for valid datasets.  What is the motivation?